### PR TITLE
[SW-1045, SW-1160, SW-1170, 1169] Implement input validation for Bitcoin addresses.

### PR DESCRIPTION
**Pull request by konradodo**

### CODE REVIEW

File: package-lock.json
Comment (pstachel) [inline package-lock.json:1]: so many changes just from one library?

File: src/protocols/bitcoin/components/TransferSendForm.vue [hunk 1]
```diff
@@ -9,12 +9,11 @@
     class="transfer-send-form"
   >
     <template #recipient>
-      <!--      TODO - set validation rules -->
       <TransferSendRecipient
         v-model.trim="formModel.address"
         :placeholder="$t('modals.send.recipientPlaceholderBtc')"
         :errors="errors"
-        :validation-rules="{}"
+        :validation-rules="{ token_to_an_btc_address: formModel.address }"
```
Comment (pstachel) [inline src/protocols/bitcoin/components/TransferSendForm.vue:16]: I feel like you could get rid of `an` from the validation rule

File: src/store/plugins/veeValidate.js [hunk 2]
```diff
@@ -124,6 +126,9 @@ export default (store) => {
     ),
     { params: ['isToken'] });
 
+  defineRule('token_to_an_btc_address',
+    (value) => btcValidate(value));
```
Comment (pstachel) [inline src/store/plugins/veeValidate.js:130]: does the line need to be split?

**Review by subhod-i — APPROVED** [review]
LGTM. The pipeline is failing!

File: src/popup/locales/en.json [hunk 3]
```diff
@@ -1156,6 +1156,7 @@
     "nameUnregistered": "This name has already been registered.",
     "notSameAs": "Sender and receiver address are the same. You will send tokens to your own account.",
     "tokenToAnAddress": "Unfortunately you can only send AEX-9 Tokens to an address",
+    "tokenToAnBtcAddress": "Invalid BTC address",
```
Comment (peronczyk) [inline src/popup/locales/en.json:1159]: I suggest to not use any protocol specific translations. Maybe you can use dynamic properties?

File: src/store/plugins/veeValidate.js [hunk 4]
```diff
@@ -124,6 +126,9 @@ export default (store) => {
     ),
     { params: ['isToken'] });
 
+  defineRule('token_to_an_btc_address',
```
Comment (peronczyk) [inline src/store/plugins/veeValidate.js:129]: Why not just `address_btc`? The validation checks if the address string is a valid BTC address.

File: package.json [hunk 5]
```diff
@@ -41,6 +41,7 @@
     "@zxing/library": "^0.19.1",
     "bignumber.js": "^9.0.2",
     "bip32": "^4.0.0",
+    "bitcoin-address-validation": "^2.2.3",
```
Comment (CedrikNikita) [inline package.json:44]: Maybe it is enough to use this regex and not to bring the external library?
`public addressValidationPattern: string = '^(?:[13]{1}[a-km-zA-HJ-NP-Z1-9]{25,34}|bc1[a-z0-9]{39,59})$'`
https://github.com/airgap-it/airgap-coin-lib/blob/912f326124ab9928bcc6b23f63cca55919a5b655/packages/bitcoin/src/v0/protocol/BitcoinProtocol.ts#L210

Comment (konradodo) [reply package.json]: you have right, regex is enough

File: src/popup/locales/en.json (hunk 3 above)
Comment (konradodo) [reply src/popup/locales/en.json]: BTC validation check only address, in AE is checking also the .chainame. I think this should be separated.

File: src/protocols/bitcoin/helpers/index.ts [hunk 6]
```diff
@@ -1,5 +1,11 @@
 import { BTC_COIN_PRECISION } from '../config';
 
+const btcRegExp = new RegExp('^(?:[13]{1}[a-km-zA-HJ-NP-Z1-9]{25,34}|bc1[a-z0-9]{39,59})$');
+
 export function satoshiToBtc(amount: number) {
   return amount / 10 ** BTC_COIN_PRECISION;
 }
+
+export function validBtcAddress(address: string) {
```
Comment (CedrikNikita) [inline src/protocols/bitcoin/helpers/index.ts:9]: I suggest to change the function name to `isBtcAddressValid`

**Review by CedrikNikita — APPROVED** [review]
Code changers look reasonable. I didn't test it.

**Review by CedrikNikita — CHANGES_REQUESTED** [review]
For me it doesn't work. I used an address account from one of the created account and set to another.

**Comment by Liubov-crypto** [comment]
LGTM

**Review by CedrikNikita — APPROVED** [review]

**Pull request merged**